File: framed-reflection/framedReflection.js
const mirror = (string) => {

    let longestIndex = 0;
    const split = string.split(' '); //split on word, not letter

    split.forEach((element) => {
        if (element.length > longestIndex){
            longestIndex = element.length;
        };
    });

    let output = [];

        for(let i = 0; i < split.length; i++){ //loop through sentence / word by each word
            let reverse = split[i].split('').reverse().join(''); //reverses the word
            if(reverse.length < longestIndex){ 
                let numberOfSpaces = longestIndex - reverse.length;
                let spaces = ` `.repeat(numberOfSpaces);
                reverse = reverse.concat(spaces); //fills spaces if needed
            };
            output.push(`* ${reverse} *\n`);
            if(i === split.length - 1){
                output.unshift('\n');
            };
        };

    const frame = '*'.repeat(longestIndex + 4);
    output.unshift(frame);
    output.push(frame);

    return output.join('');
};

module.exports = { mirror };

//https://www.codewars.com/jata/581331293788bc1702001fa6/train/javascript

File: python-venv/framed_reflection/kata_function/framed_reflection.py
def mirror(string):

    split = string.split(' ')
    longestIndex = len(max(split, key=len))
    
    output = ['\n']
    for i in range(0, len(split), 1):
        reverse = split[i][::-1]
        if(len(reverse) < longestIndex):
            numberOfSpaces = longestIndex - len(reverse)
            spaces = ' ' * numberOfSpaces
            reverse = reverse + spaces
        output.append('* ' + reverse + ' *\n')

    frame = '*' * longestIndex + '****'
    output.insert(0, frame)
    output.append(frame)

    return ''.join(output)

#https://www.codewars.com/kata/581331293788bc1702001fa6/train/python